### ISSUE TITLE
Unable to reproduce amazing results using the inference.py script

### ISSUE THREAD

**Issue by gohar-malik** [webtoon/matteformer#3]
Hi Author @pgt4861,
Thanks for your work. I am a big fan of your paper and was really impressed by how well it uses the trimap information using prior tokens. 
I was experimenting with the code but was unable to reproduce the results in terms of quality. Below are some of the examples that I ran.
In each row, from left to right, we have the original image, the trimap, the alpha mask predicted by matteformer and the final matted image.
As you can see, the output is not as good as the model can perform. Specifically in the unknown regions of the trimap, like around the fingers of the boy, the feet of the horse and the neck of the table lamp. 
Can you please guide me how to improve these results? 
P.S: I am generating these trimaps automatically from a segmentation mask using the Euclidean distance transform.

![avery](https://user-images.githubusercontent.com/35797168/175934551-135acf3f-66de-4e74-a485-1609b45af911.png)
![necklace_gold](https://user-images.githubusercontent.com/35797168/175934564-aeafee51-f61c-4a95-aa4b-868e3647541d.png)
![horse](https://user-images.githubusercontent.com/35797168/175936138-eed79154-4987-4e07-8a5f-1bf06bd1c70a.png)
![lamp2_meitu_1](https://user-images.githubusercontent.com/35797168/175936142-7234831f-7acd-4500-841a-ccce35bbe142.png)

**Comment by hackkhai** [comment]
Hey, @gohar-malik i can help you with this contact me at khailashsanthakumar@gmail.com

**Comment by pgt4861** [comment]
Hi @gohar-malik 
Thank you for you interest on our work.
I think there are some ways to improve the results on real-world applications.

First, there are some domain gaps between real world image and composited images (train data). So if the model is so overfitted to the composited (synthetic) data, it may work not well on the real-world images.
It seems that some augmentation method can be helpful to deal with the problem.

Second, the number of foreground objects in the train data is not enough to cover the diversity of real world objects.
So, I recommend you to collect foregrounds (with good ground truth) as many as possible for your target problem.

Thank you
